server/server_test.go

```diff
@@ -12,205 +12,325 @@ import (
-			randomOccurrence                      *grafeas_proto.Occurrence
-			grafeasBatchCreateOccurrencesRequest  *grafeas_proto.BatchCreateOccurrencesRequest
-			grafeasBatchCreateOccurrencesResponse *grafeas_proto.BatchCreateOccurrencesResponse
-		)
+	When("server is initialized", func() {
+		It("checks if rode project exists", func() {
+			grafeasProjectsClient.
+				EXPECT().
+				GetProject(gomock.AssignableToTypeOf(context.Background()), gomock.Eq(getProjectRequest))
 
-		BeforeEach(func() {
-			randomOccurrence = createRandomOccurrence(grafeas_common_proto.NoteKind_NOTE_KIND_UNSPECIFIED)
-			// expected Grafeas BatchCreateOccurrences request
-			grafeasBatchCreateOccurrencesRequest = &grafeas_proto.BatchCreateOccurrencesRequest{
-				Parent: "projects/rode",
-				Occurrences: []*grafeas_proto.Occurrence{
-					randomOccurrence,
-				},
-			}
-			// mock Grafeas BatchCreateOccurrences response
-			grafeasBatchCreateOccurrencesResponse = &grafeas_proto.BatchCreateOccurrencesResponse{
-				Occurrences: []*grafeas_proto.Occurrence{
-					randomOccurrence,
-				},
-			}
+			rodeServer, rodeServerError = NewRodeServer(logger.Named("rode server test"), GrafeasClients{grafeasClient, grafeasProjectsClient})
 		})
 
-		It("should forward the batch create occurrence request to grafeas", func() {
-			// ensure Grafeas BatchCreateOccurrences is called with expected request and inject response
-			grafeasClient.EXPECT().BatchCreateOccurrences(gomock.AssignableToTypeOf(context.Background()), gomock.Eq(grafeasBatchCreateOccurrencesRequest)).Return(grafeasBatchCreateOccurrencesResponse, nil)
+		When("rode project does not exists", func() {
+			var (
+				createProjectRequest = &grafeas_project_proto.CreateProjectRequest{
+					Project: &grafeas_project_proto.Project{Name: "projects/rode"},
+				}
+			)
+			BeforeEach(func() {
+				grafeasProjectsClient.
+					EXPECT().
+					GetProject(gomock.AssignableToTypeOf(context.Background()), gomock.Eq(getProjectRequest)).
+					Return(nil, status.Error(codes.NotFound, "Not found"))
+			})
 
-			batchCreateOccurrencesRequest := &pb.BatchCreateOccurrencesRequest{
-				Occurrences: []*grafeas_proto.Occurrence{
-					randomOccurrence,
-				},
-			}
+			It("creates rode project", func() {
+				grafeasProjectsClient.
+					EXPECT().
+					CreateProject(gomock.AssignableToTypeOf(context.Background()), gomock.Eq(createProjectRequest))
 
-			response, err := rodeServer.BatchCreateOccurrences(context.Background(), batchCreateOccurrencesRequest)
-			Expect(err).ToNot(HaveOccurred())
+				rodeServer, rodeServerError = NewRodeServer(logger.Named("rode server test"), GrafeasClients{grafeasClient, grafeasProjectsClient})
+			})
+
+			When("create project returns error from Grafeas", func() {
+				BeforeEach(func() {
+					grafeasProjectsClient.
+						EXPECT().
+						CreateProject(gomock.AssignableToTypeOf(context.Background()), gomock.Eq(createProjectRequest)).Return(nil, status.Error(codes.Internal, "CREATE_PROJECT_ERROR"))
```

Comment:
can we create or use an existing constant (if one exists) for both `CREATE_PROJECT_ERROR` and `GET_PROJECT_ERROR`?